daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -696,17 +702,22 @@ private[lf] final class Compiler(
       )
     }
 
-  private[this] val IdentityDef = SDefinition(pipeline(fun1((pos, env) => env.toSEVar(pos))))
+  private[this] val UnitDef = SDefinition(t.SEValue.Unit)
 
-  // Turn a template value into an interface value. Since interfaces have a
-  // toll-free representation (for now), this is just the identity function.
-  // But the existence of ImplementsDefRef implies that the template implements
-  // the interface, which is useful in itself.
+  // Witness the fact that the template 'tmplId' implements the interface 'ifaceId'
   private[this] def compileImplements(
```

Comment:
Just for my own education: Why do we do this slightly weird implementation where we generate dummy definitions with no meaningful value. Afaict, that information is fully contained in the `PackageInterface` already. Is this supposed to make the lookup more efficient? Does it actually make a difference in practice? If so, maybe some cache as a separate field in `CompiledPackages` would be less confusing?

Reply:
it comes from one of the first interfaces PRs, https://github.com/digital-asset/daml/pull/10911. I think the original idea was that we'd have different conversion functions per (template, interface), and those would live at `ImplementsDefRef(template,interface)`. What actually happened was that we went with the uniform `AnyTemplate` representation, so we ended up never using the functions at `ImplementsDefRef(template,interface)`, just their presence/absence. 

I don't think there was an efficiency argument, just the potential of directly addressing the conversion functions. Since that's not where we went, we could implement your suggested refactor in a separate PR